msom/src/main/java/pl/edu/uj/fais/wpz/msom/dao/ModuleDaoImpl.java

```diff
@@ -29,5 +30,11 @@
         modulesByControllerUnitQuery.setParameter("id", controllerUnit.getId());
         return modulesByControllerUnitQuery.list();
     }
+    
+    @Override
+    public List<Module> findAll() {
+        return getCurrentSession().createCriteria(daoType).
+                setResultTransformer(Criteria.DISTINCT_ROOT_ENTITY).list();
```

Comment:
hah, weszlam glebiej ale jak nie probowalam poprawic Module entity ale bezskutecznie wiec sie zdenerwowalam i wyszlo to co wyzej.